digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala

```diff
@@ -2212,52 +2222,37 @@ object DataFixture {
+      println(s"Try to solve ${incompleteVKMInfo.size}")
 
-      val frozenAddresses = frozenRoadLinks.flatMap { frozen =>
-        val (first, last) = GeometryUtils.geometryEndpoints(frozen.geometry)
+      val adjRoadLinks = incompleteVKMInfo.flatMap(x => roadLinkService.getAdjacent(x.linkId, newTransaction = false))
 
-        try {
-          val address = geometryTransform.vkmGeometryTransform.coordsToAddresses(Seq(first, last), includePedestrian = Some(true))
-          if (address.isEmpty || (address.nonEmpty && address.size != 2)) {
-            println("Problems in addresses")
-            Seq()
-          } else {
-            val grouped = address.groupBy(addr => (addr.road, addr.roadPart))
-            if (grouped.keys.size > 1) {
-              val recalculateAddresses = recalculateAddress(frozen, mappedAddresses)
-              if (recalculateAddresses.size == 1)
-                Some(recalculateAddresses.head)
-              else {
-                recalculateAddresses.foreach { recalc =>
-                  println(s" more than one road -> linkId: ${recalc.linkId} road ${recalc.road} roadPart ${recalc.roadPart} track ${recalc.track}  etays ${recalc.startAddressM} let ${recalc.endAddressM} start ${recalc.startMValue}  end let ${recalc.endMValue} ")
-                }
-              }
-              None
-            } else {
-              val orderedAddress = address.sortBy(_.addrM)
-              Some(RoadAddressTEMP(frozen.linkId, orderedAddress.head.road, orderedAddress.head.roadPart, Track.Unknown, orderedAddress.head.addrM, orderedAddress.last.addrM, 0, GeometryUtils.geometryLength(frozen.geometry), frozen.geometry, municipalityCode = Some(frozen.municipalityCode)))
-            }
-          }
-        } catch {
-          case ex: Exception =>
-            println(s"Exception in VKM for linkId ${frozen.linkId}")
-            None
+      val roadLinks = roadLinkService.getRoadLinksAndComplementariesFromVVH(incompleteVKMInfo.map(_.linkId).toSet , newTransaction = false) ++ adjRoadLinks
+
+      val allRoadAddress = roadAddressService.getAllByLinkIds(roadLinks.map(_.linkId))
+      val vkmRoadAddress = roadLinkTempDao.getByLinkIds(adjRoadLinks.map(_.linkId).toSet).filterNot(_.track == Track.Unknown)
+
+      val mappedAddresses = allRoadAddress.map { address =>
+        RoadAddressTEMP(address.linkId, address.roadNumber, address.roadPartNumber, address.track, address.startAddrMValue, address.endAddrMValue, address.startMValue, address.endMValue, address.geom, Some(address.sideCode))
+      } ++ vkmRoadAddress
+
+      val result = recalculateTrackAndSideCode(mappedAddresses, incompleteVKMInfo, incompleteVKMInfo.size, roadLinks.filter(_.administrativeClass == State), Seq())
+      val linkTypesCombined = Seq(CycleOrPedestrianPath, PedestrianZone, CableFerry)
+
+      val toCreate = result.flatMap(_.possibleToCreate).map { roadAddr =>
+        val roadLink = roadLinks.find(_.linkId == roadAddr.linkId).get
```

Comment:
In case of doesn't find any road link this will fail

Reply:
Will exist.
Because in line 2239 we are passing the roadLinks that are state to the function recalculateTrackAndSideCode.

So when we will try to find the roadlinks in line 2243 itshould exist